### ISSUE TITLE
Consider HTML preprocessing?

### ISSUE THREAD

**Issue by Taylord93** [TricomB2B/ng-boilerplate#10]
Should we consider switching to a preprocessor (I like [Jade](http://jade-lang.com/), [Haml](http://haml.info/) is another good option, although intended for Ruby projects) to see if we can speed up dev efforts a little when it comes to making templates?

**Comment by Taylord93** [comment]
I guess Jade has been moved and renamed to [Pug](https://github.com/pugjs/pug) and is now Node-based.

**Comment by Toddses** [comment]
I've never been a big fan of Jade. I don't write Jade any faster than HTML, so for me that idea of speeding up dev time just wasn't there. I never really thought it was that different from HTML either, so I never saw the point in adding an extra layer of complexity for something that I saw as adding very little value. I'm open to it, though, if there's a compelling reason for it, as I haven't revisited it in some time.

**Comment by Taylord93** [comment]
Where I find it to save time is mostly that you dont have to write closing brackets (which isn't an issue regardless if your ide does it for you). Sounds simple but overtime does save quite a bit of time. You also dont have to write out class="" instead, you write div.className#id(attribute) and etc. 

Some other pluses are (arguably) readability. Jade does seem a bit cleaner, but I am not really for or against it strongly. Maybe we should just set up a branch and do a test run?

**Comment by Toddses** [comment]
Test branch sounds like a plan.

A couple other thoughts to keep in mind. Adding in Jade would be an additional thing someone coming into a project would have to learn (a contractor or new hire). I don't think it's hard to learn, just something to keep in mind. Another is that Jade is basically a templating language and has features that are related to that. Angular's templating pretty much offers many of the same features. We'd really need to hammer out a set of rules for using them together to stay consistent.

**Comment by Taylord93** [comment]
This is true. I say we give it a test run with a future project and see how it goes. If it does more harm than good, we will kick it off the table. Im not dead set on using it or not, but its worth a shot IMO.
